inform7/Internal/Inter/Architecture32Kit/Sections/Input Output.i6t

```diff
@@ -319,6 +321,12 @@ to document all of that.
         glk_put_char_stream(gg_commandstr, 10); ! newline
     }
   .KPContinue;
+    for ( ix = 1 : ix <= (a_buffer-->0) : ix++ )
+      if (((a_buffer-->ix) < 32) || ((((a_buffer-->ix) >= 8192) && ((a_buffer-->ix) <= 8202))))
+        a_buffer-->ix = 32;
+      else
+        for ( jy = 1 : jy <= (UnicodeWhitespace-->0) : jy++ )
+          if ((a_buffer-->ix) == (UnicodeWhitespace-->jy)) a_buffer-->ix = 32;
```

Comment:
If you're seriously worried about number of Glulx opcodes executed here (maybe you shouldn't), then this could be a lot more efficient. As written, almost all characters, including any ASCII characters > 0x20, will enter this `else` and go through a couple dozen instructions to be compared against every element in `UnicodeWhitespace`. A hard-coded decision tree with nested `if`s could e.g. skip all code points between 0x20 and 133 after two comparisons. Alternatively, keeping the full list of code points from the previous commit and doing a binary search over it needs at most six comparisons (and very little other work, if the "binary search loop" is fully unrolled instead of being written as a real loop) per code point in `a_buffer`.

Reply:
Actually, you could just use Glulx's `@binarysearch` opcode to get the whole search done quickly and avoid having to write + debug any sort of decision tree or binary search yourself.

Reply:
@hanna-kruppe Or would it be better to `@linearsearch` the input for each white space character?

Reply:
Yeah, I don't think anything that does multiple passes through the buffer (even if all but one of them happen via `linearsearch`) will be faster *or* simpler than just doing a single pass with `binarysearch` handling the "should this code point be replaced with U+0020?" check. If this was native code with access to SIMD instructions, I could imagine different strategies to carve out "fast paths" if necessary, but that's not an option here and probably overkill anyway.